Improve getting scheme from xcuserdata if not found at xcshareddata

matyashlavacek opened the pull request:
Prior to v2.0.2 and Xcode 7.3 slather was working fine with our project, after the changes in Xcode and the fixes in slather our project started to fail on "No such file or directory" as slather was looking only into the xcshareddata directory for the scheme in xcodeproj and we don't have the scheme shared yet. 
I have modified the code to also look into the xcuserdata directory in case it cannot find the scheme in xcshareddata, which seems logical to me. 
Was there any specific reason to look just into the shared dir for the scheme?

neonichu commented on lib/slather/project.rb:
```diff
@@ -328,7 +328,14 @@ def find_binary_file
-        raise StandardError, "No shared scheme named '#{self.scheme}' found in #{self.path}" unless File.exists? xcscheme_path
+        # Try to look inside 'xcuserdata' if the scheme is not found in 'xcshareddata'
+        if !File.file?(xcscheme_path)
+          puts "No shared scheme named '#{self.scheme}' found in #{schemes_path}. Trying to look into xcuserdata."
```
Please remove this.

neonichu commented:
I think the reason was just that most users will be running this on CI where probably no user schemes will exist. Thanks for the contribution, if you could address the one comment and add a changelog entry, it should be good to merge.

matyashlavacek commented:
Thanks for the comments, I've updated the changelog and removed the line you mentioned.

_And sorry for the multiple commits, I'm still fairly new to contributing to public repo_

neonichu commented:
That's fine, thanks :) I squashed and merged this manually in 1c0eec6
